swiftnav/src/reference_frame.rs

```diff
@@ -0,0 +1,656 @@
+use crate::coords::{Coordinate, ECEF};
+use std::{convert::TryFrom, fmt, str::FromStr};
+
+/// Reference Frames
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+pub enum ReferenceFrame {
+    ITRF2008,
+    ITRF2014,
+    ITRF2020,
+    ETRF2008,
+    ETRF2014,
+    ETRF2020,
+    /// i.e. NAD83(2011)
+    NAD83_2011,
+    /// i.e. NAD83(CSRS) - Canadian Spatial Reference System
+    #[allow(non_camel_case_types)]
+    NAD83_CSRS,
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
+pub struct InvalidReferenceFrameName(String);
+
+impl fmt::Display for InvalidReferenceFrameName {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Invalid reference frame name ({})", self.0)
+    }
+}
+
+impl std::error::Error for InvalidReferenceFrameName {}
+
+impl TryFrom<&str> for ReferenceFrame {
+    type Error = InvalidReferenceFrameName;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        match value {
+            "ITRF2008" => Ok(ReferenceFrame::ITRF2008),
+            "ITRF2014" => Ok(ReferenceFrame::ITRF2014),
+            "ITRF2020" => Ok(ReferenceFrame::ITRF2020),
+            "ETRF2008" => Ok(ReferenceFrame::ETRF2008),
+            "ETRF2014" => Ok(ReferenceFrame::ETRF2014),
+            "ETRF2020" => Ok(ReferenceFrame::ETRF2020),
+            "NAD83(2011)" => Ok(ReferenceFrame::NAD83_2011),
+            "NAD83_2011" => Ok(ReferenceFrame::NAD83_2011),
+            "NAD83(CSRS)" => Ok(ReferenceFrame::NAD83_CSRS),
+            "NAD83_CSRS" => Ok(ReferenceFrame::NAD83_CSRS),
+            _ => Err(InvalidReferenceFrameName(value.to_string())),
```

Comment:
I think it might be a good use for strum and EnumString here: https://docs.rs/strum_macros/latest/strum_macros/derive.EnumString.html